From 0204e17bc656e6c59b4010283e67f9251eb12c41 Mon Sep 17 00:00:00 2001
From: Dan Fandrich <dan@coneharvesters.com>
Date: Wed, 23 Apr 2014 11:01:30 +0200
Subject: [PATCH] cyassl: Use error-ssl.h when available

Versions since at least 2.9.4 renamed error.h to error-ssl.h, so use
whichever one is available.
---
 configure.ac      |    3 +++
 lib/vtls/cyassl.c |    9 +++++++--
 2 files changed, 10 insertions(+), 2 deletions(-)

--- a/configure.ac
+++ b/configure.ac
@@ -2003,6 +2003,9 @@ if test "$curl_ssl_msg" = "$init_ssl_msg
       dnl cyassl/ctaocrypt/types.h needs SIZEOF_LONG_LONG defined!
       AC_CHECK_SIZEOF(long long)
 
+      dnl Versions since at least 2.9.4 renamed error.h to error-ssl.h
+      AC_CHECK_HEADERS(cyassl/error-ssl.h)
+
       LIBS="-lcyassl -lm $LIBS"
 
       if test -n "$cyassllib"; then
--- a/lib/vtls/cyassl.c
+++ b/lib/vtls/cyassl.c
@@ -47,11 +47,16 @@
 #define _MPRINTF_REPLACE /* use our functions only */
 #include <curl/mprintf.h>
 #include "curl_memory.h"
-/* The last #include file should be: */
-#include "memdebug.h"
+
 #include <cyassl/ssl.h>
+#ifdef HAVE_CYASSL_ERROR_SSL_H
+#include <cyassl/error-ssl.h>
+#else
 #include <cyassl/error.h>
+#endif
 
+/* The last #include file should be: */
+#include "memdebug.h"
 
 static Curl_recv cyassl_recv;
 static Curl_send cyassl_send;
